src/server.cpp

```diff
@@ -481,8 +481,15 @@ void Server::step(float dtime)
 	}
 	// Throw if fatal error occurred in thread
 	std::string async_err = m_async_fatal_error.get();
-	if(async_err != ""){
+	if(async_err != "") {
+#ifdef SERVER
+		errorstream << "UNRECOVERABLE error occurred. Stopping server. "
+				<< "Please fix the following error:" << std::endl
+				<< async_err << std::endl;
+		assert(false);
```

Comment:
assert() in release builds is a no-op and does nothing at all.

Perhaps FATAL_ERROR() would be more appropriate?